test/factories/ansible_roles.rb

```diff
@@ -4,4 +4,16 @@
   factory :ansible_role do
     sequence(:name) { |n| "ansible_role_#{n}" }
   end
+
+  factory :host_ansible_role do
+    sequence(:position)
+    host_id { nil }
+    ansible_role_id { nil }
```

Comment:
Why don't we use `host` and `ansible_role`, it would have the benefit of autocreating those associated records if you don't specify them and its more natural to FactoryBot, so should be easier to read and use.